build.gradle

```diff
@@ -20,21 +13,77 @@ buildscript {
     }
 }
 
-dependencies {
-    compile "org.elasticsearch:elasticsearch:7.5.1"
-    compile "org.elasticsearch.plugin:parent-join-client:7.5.1"
-    compile "org.jetbrains.kotlin:kotlin-stdlib:1.3.61"
-    testCompile "org.apache.logging.log4j:log4j-api:2.13.0"
-    testCompile "org.apache.logging.log4j:log4j-core:2.13.0"
-    testCompile 'com.fasterxml.jackson.core:jackson-databind:2.10.2'
-    testCompile "junit:junit:4.13"
-}
+group 'mbuhot'
+version '1.0.0beta1'
+
+apply plugin: 'java-base'
+apply plugin: 'kotlin'
+//apply plugin: 'com.jfrog.bintray'
+apply plugin: 'maven-publish'
 
 repositories {
     mavenCentral()
+
+    maven {
+        def repo_url = project.hasProperty('repo_url') ? project.property('repo_url')
+                : 'https://nexus.semandex.net:44355/repository/semandex-maven-prod/'
+        url repo_url
+        allowInsecureProtocol = true
+    }
 }
 
-task sourcesJar(type: Jar) {
+boolean canNexus = true
+Set<String> missingProps = []
+
+if (!hasProperty( 'nexus_user' )) {
+    canNexus = false
+    missingProps.add( 'nexus_user' )
+}
+if (!hasProperty( 'nexus_pass' )) {
+    canNexus = false
+    missingProps.add( 'nexus_pass' )
+}
+
+if (!canNexus) {
+    if ( gradle.startParameter.taskNames.any {it.contains( 'publish' ) } ) {
+        logger.warn("Will not publish to nexus repo due to missing credentials: " + missingProps)
+    }
+    return
+}
+
+publishing {
+    repositories {
+        maven {
+            url hasProperty( 'nexus_repo' ) ? property('nexus_repo')
+                    : 'https://nexus.semandex.net:44355/repository/semandex-maven-hosted/'
+            allowInsecureProtocol = true
+            credentials {
+                username = nexus_user
+                password = nexus_pass
+            }
+        }
+    }
+}
+
+
+ext {
+  kotlinVersion = '1.3.61'
```

Comment:
Isnt this an old version of kotlin? We cant upgrade right now?

Reply:
no. only with v2 of opensearch